linalg/solvers.cpp

```diff
@@ -861,7 +861,7 @@ void CGSolver::Mult(const Vector &b, Vector &x) const
       }
       nom = betanom;
    }
-   if (print_options.first_and_last)
+   if (print_options.first_and_last && !print_options.iterations)
```

Comment:
Prevent double-printing the last iteration when both `first_and_last` and `iterations` are enabled.